CorsixTH/Lua/humanoid_actions/seek_room.lua

```diff
@@ -48,12 +48,22 @@ function SeekRoomAction:setDiagnosisRoom(room)
   return self
 end
 
+--! Finds a relevant diagnosis/treatment room(s) for a patient
+--! If diagnosis room, attempt to use GP's choice first, else select any other available
+--! room at random.
 local action_seek_room_find_room = permanent"action_seek_room_find_room"( function(action, humanoid)
   local room_type = action.room_type
   if action.diagnosis_room then
+    -- Attempt to use the GP's chosen diagnosis room first
+    local room
```

Comment:
No need to expand room's scope like this, you can have two `local room` in the same function when the scope doesn't overlap.

Reply:
You could try moving it to line 60 to avoid the warning perhaps?

Reply:
Correct.

`room_at_index` declaration could also be moved more inward.